pkg/controller/secret/secret_controller.go

```diff
@@ -0,0 +1,428 @@
+func addSnitchSecretToAlertManagerConfig(r *ReconcileSecret, request *reconcile.Request, amconfig *alertmanager.Config, snitchsecret string) {
+
+	// Define the contents of the WebhookConfig which is part of the Watchdog receiver.
+	// The Watchdog receiver uses the Dead Man's Snitch external service as its webhook.
+	snitchconfig := &alertmanager.WebhookConfig{
+		NotifierConfig: alertmanager.NotifierConfig{VSendResolved: true},
+		URL:            snitchsecret,
+	}
+
+	// Overwrite the existing Watchdog config with the updated version specified above.
+	// This keeps other receivers intact while updating only the Watchdog receiver.
+	watchdogabsent := true
+	for i, receiver := range amconfig.Receivers {
+		fmt.Println("DEBUG: Found Receiver named:", receiver.Name)
+		if receiver.Name == "watchdog" {
+			fmt.Println("DEBUG: Overwriting watchdog receiver:", receiver.Name)
+			amconfig.Receivers[i].WebhookConfigs = []*alertmanager.WebhookConfig{snitchconfig}
+			watchdogabsent = false
+		} else {
+			fmt.Println("DEBUG: Skipping Receiver named", receiver.Name)
+		}
+	}
+
+	// Create the Watchdog receiver if it doesn't already exist.
+	if watchdogabsent {
+		fmt.Println("DEBUG: Watchdog receiver is absent. Creating new receiver.")
+		newreceiver := &alertmanager.Receiver{
+			Name:           "watchdog",
+			WebhookConfigs: []*alertmanager.WebhookConfig{snitchconfig},
+		}
+		amconfig.Receivers = append(amconfig.Receivers, newreceiver)
+	}
+
+	// Create a route for the new Watchdog receiver.
+	wdroute := &alertmanager.Route{
+		Receiver:       "watchdog",
+		RepeatInterval: "5m",
+		Match:          map[string]string{"alertname": "Watchdog"},
+	}
+
+	// Insert the Route for the Watchdog Receiver.
+	routeabsent := true
+	for i, route := range amconfig.Route.Routes {
+		fmt.Println("DEBUG: Found Route for Receiver:", route.Receiver)
+		if route.Receiver == "watchdog" {
+			fmt.Println("DEBUG: Overwriting Watchdog Route for Receiver:", route.Receiver)
+			amconfig.Route.Routes[i] = wdroute
+			routeabsent = false
+		} else {
+			fmt.Println("DEBUG: Skipping Route for Receiver named", route.Receiver)
+		}
+	}
+
+	// Create Route for Watchdog Receiver if it doesn't already exist.
+	if routeabsent {
+		fmt.Println("DEBUG: Route for Watchdog Receiver is absent. Creating new Route.")
+		amconfig.Route.Routes = append(amconfig.Route.Routes, wdroute)
+	}
+}
+
+// removeFromReceivers removes the specified index from a slice of Receivers.
+func removeFromReceivers(slice []*alertmanager.Receiver, i int) []*alertmanager.Receiver {
+	// copy all the elements in the slice except for the one we want to delete
+	slicecopy := []*alertmanager.Receiver{}
+	for index, receiver := range slice {
+		if index != i {
+			fmt.Println("DEBUG: The index", index, "doesn't match the element we want to delete:", i, ". So copy it over to slicecopy")
+			slicecopy = append(slicecopy, receiver)
+			fmt.Println("DEBUG: contents of slicecopy so far:")
+			for _, x := range slicecopy {
```

Comment:
I don't think you need this function here It just prints out the data and isn't modifying anything. Is there a reason this needs to be logged ? Also if you want to delete an item from an array it is easier to do something like this 

```
a = append(a[:i], a[i+1:]...) ---> This creates a copy of the array from 0:i and from i+:(end)
// or
a = a[:i+copy(a[i:], a[i+1:])]
```

where i is the index you want to delete. 

Let me know if you have any questions

Reply:
Thanks, that was some extra debug output that I don't need anymore. I was having trouble deleting Receivers/Routes from the Alertmanager config. I'll remove that.

Reply:
Ok, this is fixed and tested now.